samcli/lib/pipeline/bootstrap/stage.py

```diff
@@ -100,14 +136,64 @@ def __init__(
         )
         self.color = Colored()
 
+        self.oidc_provider: OidcProvider = OidcProvider(
+            client_id=oidc_client_id,
+            provider_url=oidc_provider_url,
+            thumbprint="",
+            comment="IAM OIDC Identity Provider",
+            arn="",
+        )
+
+    def _should_create_new_provider(self) -> bool:
+        if not self.oidc_provider.provider_url:
+            return False
+        iam_client = boto3.client("iam")
+        providers = iam_client.list_open_id_connect_providers()
+        url_to_compare = self.oidc_provider.provider_url.replace("https://", "")
```

Comment:
Yea since the ARN has the entire URL except for the protocol, and then by seeing if that part is in the ARN of one of the existing providers we can figure out if a new provider needs to be created or not without having to ask.

arn:aws:iam::<ACCOUNT>:oidc-provider/api.bitbucket.org/2.0/workspaces/<USERNAME>/pipelines-config/identity/oidc

is the format of the ARN that using the public bitbucket instance for OIDC ends up as.

Reply:
Oh gotcha, makes sense. A comment with e.g. example ARN would be helpful; it's not immediately obvious why a URL is compared with an ARN.